Snabble/UI/Checkout/InflightCheckoutContinuationViewController.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  InflightCheckoutContinuationViewController.swift
+//  Snabble
+//
+//  Created by Gereon Steffens on 07.04.22.
+//
+
+import UIKit
+
+public final class InflightCheckoutContinuationViewController: UIViewController {
+    private let spinner = UIActivityIndicatorView()
+    private let label = UILabel()
+
+    public weak var paymentDelegate: PaymentDelegate?
+    public var shoppingCart: ShoppingCart?
+
+    override public func viewDidLoad() {
+        super.viewDidLoad()
+
+        hidesBottomBarWhenPushed = true
+        navigationController?.setNavigationBarHidden(true, animated: false)
+        title = L10n.Snabble.Payment.confirm
+
+        view.backgroundColor = .systemBackground
+
+        if #available(iOS 13.0, *) {
+            spinner.style = .medium
+        } else {
+            spinner.style = .gray
+        }
+        spinner.startAnimating()
+
+        label.text = L10n.Snabble.PaymentContinuation.message
+        label.numberOfLines = 0
+        label.useDynamicFont(forTextStyle: .body)
+
+        let stackview = UIStackView(arrangedSubviews: [ spinner, label ])
+        stackview.translatesAutoresizingMaskIntoConstraints = false
+        stackview.axis = .vertical
+        stackview.spacing = 16
+        stackview.alignment = .center
+
+        view.addSubview(stackview)
+
+        NSLayoutConstraint.activate([
+            stackview.centerYAnchor.constraint(equalTo: view.centerYAnchor),
+            stackview.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 16),
+            stackview.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -16)
+        ])
+
+        if let inFlightCheckout = Snabble.inFlightCheckout {
+            continueInFlightCheckout(with: inFlightCheckout)
+        }
```

Comment:
Sollten wir den ViewController nicht nur erzeugen wenn es ein `InFlightCheckout` gibt? Bedeutet was hältst du davon den ViewController nur zu erzeugen wenn es ein InflightCheckout gibt und diesen direkt zu übergeben?

Reply:
Genau das passiert im `HomeViewController` schon. Die Alternative wäre eine statische Factory-Methode, die hier aber einen Optional zurückgeben müsste, da bin ich kein Fan von.

Reply:
Ja, aber warum übergibst du die Variable nicht an den ViewControllern? Warum muss er selbst nochmal prüfen ob es vorhanden ist?

Reply:
Weil man dazu den State public machen müsste, und das wollte ich nicht.

Reply:
Was ist der Hintergrund? 🤔 Es löst potentiell ein Problem wenn das Parsing nicht klappt.

Reply:
Ich habe es hier mal angepasst 3482fa5ec9a9f1d5341f37d0c42d49182a2eaf9c

Mir fällt leider kein Fall ein wo es hierdurch zu Problemen kommen kann, aber in dem anderen Fall durch Parsingfehler.